### ISSUE TITLE
[BUG] MaskedDataset.__getitem__ trains MLM against input x instead of target y

### ISSUE THREAD

**Issue by ritankarsaha** [gc-os-ai/pyaptamer#461]
**Describe the bug**

In pyaptamer/datasets/dataclasses/_masked.py, the masked target tensor is cloned from x instead of y:     
                                                            
  ### Before                                                                                          
  y_masked = x.clone().detach()                                                                                 
                                                                                                                
  ### After                                                                                          
  y_masked = y.clone().detach()                                                                                 
  
  Because y_masked was built from x, the Masked Language Model pretraining objective was always comparing       
  predictions against the input sequence rather than the target sequence. For any dataset where x != y (e.g.,
  cross-modal or cross-species pairs), this makes the entire pretraining objective meaningless — the model
  learns to reconstruct its own noisy input, not the intended target. 

**Versions**
0.1.0a1

**Comment by ritankarsaha** [comment]
@NennoMP can you take a look into this issue ?

**Comment by NennoMP** [comment]
A simple keyword search on previous issues/PRs (#304, #356) would let you know that this is intended behaviour, rather than a bug. 

Please refrain from tagging maintainers and asking for a review if your PR has been opened only for a couple of hours. Give it at least a week before reaching out. Also, please refrain from feeding the codebase into LLMs to spot potntial bugs if that is what you are doing. The four new PRs opened in the last couple hours suggest that.
